src/app/components/layouts/card/terra-card.component.spec.ts

```diff
@@ -1,3 +1,185 @@
+        expect(footerElement.nativeElement.hidden).toBe(false);
+    });
+
+    it('should set class selected depending on #inputIsSelected', () =>
+    {
+        let debugElement:DebugElement = fixture.debugElement;
+        let footerElement:DebugElement = debugElement.query(By.css('div.card-footer'));
+        let terraCardElement:DebugElement = debugElement.query(By.css('div.terra-card'));
+        expect(footerElement.classes['selected']).toBeFalsy();
+        expect(terraCardElement.classes['selected']).toBeFalsy();
+
+        cardComponent.inputIsSelected = true;
+        fixture.detectChanges();
+        expect(footerElement.classes['selected']).toBeTruthy();
+        expect(terraCardElement.classes['selected']).toBeTruthy();
+    });
+
+    // header
+    it('should show div-element for header if content is given', () =>
+    {
+        let debugElement:DebugElement = fixture.debugElement;
+        let headerElement:DebugElement = debugElement.query(By.css('div.card-header'));
+        let ngContentElement:DebugElement = headerElement.query(By.css('p'));
+        expect(headerElement).toBeTruthy();
+        expect(cardComponent.viewChildHeader).toBeTruthy();
+        expect(cardComponent.viewChildHeader.nativeElement.children.length).toBeGreaterThan(0);
+        expect(ngContentElement.nativeElement.textContent.trim()).toEqual('card header');
+        expect(headerElement.nativeElement.hidden).toBe(false);
+    });
+
+    // block
+    it('should show div-element for content if content is given', () =>
+    {
+        let debugElement:DebugElement = fixture.debugElement;
+        let contentElement:DebugElement = debugElement.query(By.css('div.card-block'));
+        let ngContentElement:DebugElement = contentElement.query(By.css('p'));
+        expect(contentElement).toBeTruthy();
+        expect(ngContentElement.nativeElement.textContent.trim()).toEqual('card content');
+    });
+
+    // image
+    it('should show image if #inputImagePath is set', () =>
+    {
+        let debugElement:DebugElement;
+        let imageElement:DebugElement;
+
+        cardComponent.inputImagePath = '';
+        fixture.detectChanges();
+        debugElement = fixture.debugElement;
+        imageElement = debugElement.query(By.css('div.terra-card-image'));
+        expect(imageElement).toBeFalsy();
+
+        cardComponent.inputImagePath = expectedImagePath;
+        fixture.detectChanges();
+        imageElement = debugElement.query(By.css('div.terra-card-image'));
+        expect(imageElement).toBeTruthy();
+    });
+
+    it('should show icon if #inputPlaceholderIcon is set', () =>
+    {
+        let debugElement:DebugElement = fixture.debugElement;
+        let iconElement:DebugElement = debugElement.query(By.css('div.terra-card-placeholder'));
+        expect(iconElement).toBeTruthy();
+        expect(cardComponent.inputPlaceholderIcon).toEqual(expectedIcon);
```

Comment:
`cardComponent.inputPlaceholderIcon` in diesem Test vorher auch setzen

Reply:
done